packages/grid/x-data-grid/src/tests/filterPanel.DataGrid.test.tsx

```diff
@@ -183,11 +183,11 @@ describe('<DataGrid /> - Filter panel', () => {
 
     setColumnValue('slogan');
 
-    expect(getColumnValues(0)).to.deep.equal([]);
+    expect(getColumnValues(0)).to.deep.equal(['Nike', 'Adidas', 'Puma']);
```

Comment:
Thanks for looking into it!
While the fix is quite simple, it changed the filtering behavior in some valid use cases - this is why you had to update a few unit tests for them to pass.

Instead of always resetting the filter value, did you consider other options?
For example, resetting the filter value only if the type of the filtered column changes.
What do you think?